### PR TITLE
Accessor method implementations (continued)

### PR DESCRIPTION
**Pull request by iamshnoo**
As per the suggestions made in the last few comments in #2269 ,  I am opening this PR. As far as I can understand this only has the 7 commits with 22 files changed by me and the rest is same as the master branch of mlpack as of now. I hope I am doing this right :)

@zoq @sreenikSS If you guys find some time, could you please review the changes.

I basically did all the work in my feature branch, then  updated the master branch of my fork today, and merged the feature branch to my master. Now, I am trying to merge this master with mlpack:master.  ~~I am keeping this PR as draft for now, until it is reviewed.~~ Never mind, I am switching from draft to review mode. I guess I just wanted to use the draft feature badly :)

### CODE REVIEW

**Comment by himanshupathak21061998** [comment]
@iamshnoo Always keep the master branch for rebasing Now if you want to rebase your pr It will cause problems for you. Make sure, if you are making changes it should be on feature branch.

**Comment by iamshnoo** [comment]
@himanshupathak21061998 I am kind of new to the git workflow. I had never heard of rebasing before this. But thank you, I will keep this in mind for any new PRs I create.

File: src/mlpack/methods/ann/layer/reparametrization.hpp [hunk 1]
```diff
@@ -121,6 +121,15 @@ class Reparametrization
...
   }
 
+  //! Get the value of the stochastic parameter.
+  bool Stochastic() const { return stochastic; }
+
+  //! Get the value of the includeKl parameter.
+  bool IncludeKL() const { return includeKl; }
+
+  //! Get the value of the beta hyperparameter.
+  double Beta() const { return beta; }
```
Comment (sreenikSS) [inline src/mlpack/methods/ann/layer/reparametrization.hpp]: Nice detail with distinguishing parameter and hyperparameter :)

**Comment by sreenikSS** [comment]
@iamshnoo could you add a simple test to check if the returned values are correct?

You'll need to make changes to the file src/mlpack/tests/ann_layer_test.cpp.

File: src/mlpack/methods/ann/layer/reparametrization.hpp (hunk 1 above)
Comment (iamshnoo) [reply src/mlpack/methods/ann/layer/reparametrization.hpp]: 😅

**Comment by iamshnoo** [comment]
I am going to try some experiments with git now and in case I mess up, I will probably delete my fork and re-start. If that happens I might have to re-open this PR separately, so just a heads up about this!

And @sreenikSS sure, I will write down those tests as soon as I can. Thanks

File: src/mlpack/methods/ann/layer/highway.hpp [hunk 2]
```diff
@@ -171,6 +171,19 @@ class Highway
   //! Modify the gradient.
   OutputDataType& Gradient() { return gradient; }
 
+  //! Get the input width.
+  size_t const& InputWidth() const { return width; }
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/highway.hpp]: Some modules in mlpack only check for `InputWidth` and expect `OutputWidth` is there as well, either we remove the getter here or add a pseudo `OutputWidth` method as well.

Comment (iamshnoo) [reply src/mlpack/methods/ann/layer/highway.hpp]: I see!

**Comment by iamshnoo** [comment]
> I am going to try some experiments with git now and in case I mess up, I will probably delete my fork and re-start. If that happens I might have to re-open this PR separately, so just a heads up about this!

I messed up. I am going to delete my fork now!

**Pull request closed without merging**